Add DataFrames.jl GSOC proposal to tables.md (#1819)
JuliaLang/www.julialang.org: jsoc/gsoc/tables.md

**Expected Results:** Depends on the specific projects we would agree on.

**Mentors:** [Tanmay Mohapatra](https://github.com/tanmaykm)

## DataFrames.jl join enhancements

**Difficulty**: Hard

**Duration**: 175 hours 

[DataFrames.jl](https://github.com/JuliaData/DataFrames.jl) is one of the more popular implementations of tabular data type for Julia. One of the features it supports is data frame joining. However, more work is needed to improve this functionality. The specific targets for this project are (a final list of targets included in the scope of the project can be decided later).
@@tight-list

> * fully implement multi-threading support of joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;

Author: ```suggestion
* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```

> * fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;

Author: ```suggestion
* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies in comparison to those currently used and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```

> * implement join allowing for efficient matching on non-equal keys; special attention should be made to matching on keys that are timestamps and spatial objects;

Author: ```suggestion
* implement join allowing for efficient matching on non-equal keys; special attention should be made to matching on keys that are date/time and spatial objects;
```

**Expected Results:** Depends on the specific projects we would agree on.

**Mentors:** [Tanmay Mohapatra](https://github.com/tanmaykm)

## DataFrames.jl join enhancements

**Difficulty**: Hard

**Duration**: 175 hours 

> [DataFrames.jl](https://github.com/JuliaData/DataFrames.jl) is one of the more popular implementations of tabular data type for Julia. One of the features it supports is data frame joining. However, more work is needed to improve this functionality. The specific targets for this project are (a final list of targets included in the scope of the project can be decided later).

Reviewer: Hey @bkamins, ideally we would generally scope out these projects, the Google team advises against just including a list of open issues unless they are specific for GSoC project options, can you port over a high level work flow on a few of these?
